antenna/crashmover.py

```diff
@@ -137,102 +87,47 @@ def check_health(self, state):
-        # Indicates whether or not we're sitting on crashes to save--this helps
-        # keep Antenna alive until we're done saving crashes
-        return bool(work_to_do)
-
-    def hb_run_crashmover(self):
-        """Spawn a crashmover if there's work to do."""
-        # Spawn a new crashmover if there's stuff in the queue and we haven't
-        # hit the limit of how many we can run
-        if self.crashmover_queue and self.crashmover_pool.free_count() > 0:
-            self.crashmover_pool.spawn(self.crashmover_process_queue)
-
-    def add_crashreport(self, raw_crash, dumps, crash_id):
-        """Add a new crash report to the crashmover queue.
+    @MYMETRICS.timer("crash_handling.time")
+    def handle_crashreport(self, raw_crash, dumps, crash_id):
+        """Handle a new crash report synchronously and return whether that succeeded.
 
         :arg raw_crash: map of key/val crash annotations
         :arg dumps: map of name to memory dump
         :arg crash_id: the crash id for the crash report
 
         """
-        crash_report = CrashReport(raw_crash, dumps, crash_id)
-        crash_report.set_state(STATE_SAVE)
-
-        self.crashmover_queue.append(crash_report)
-        self.hb_run_crashmover()
-
-    def crashmover_process_queue(self):
-        """Process crashmover work.
 
-        NOTE(willkg): This has to be super careful not to lose crash reports.
-        If there's any kind of problem, this must return the crash report to
-        the relevant queue.
-
-        """
-        while self.crashmover_queue:
-            crash_report = self.crashmover_queue.popleft()
+        crash_report = CrashReport(raw_crash, dumps, crash_id)
+        for errors in range(1, MAX_ATTEMPTS + 1):
             try:
-                if crash_report.state == STATE_SAVE:
-                    # Save crash and then toss crash_id in the publish queue
-                    self.crashmover_save(crash_report)
-                    crash_report.set_state(STATE_PUBLISH)
-                    self.crashmover_queue.append(crash_report)
-
-                elif crash_report.state == STATE_PUBLISH:
-                    # Publish crash and we're done
-                    self.crashmover_publish(crash_report)
-                    self.crashmover_finish(crash_report)
-
+                self.crashmover_save(crash_report)
+                break
             except Exception:
-                MYMETRICS.incr("%s_crash_exception.count" % crash_report.state)
-                crash_report.errors += 1
+                MYMETRICS.incr("save_crash_exception.count")
                 LOGGER.exception(
-                    "Exception when processing queue (%s), state: %s; error %d/%d",
-                    crash_report.crash_id,
-                    crash_report.state,
-                    crash_report.errors,
-                    MAX_ATTEMPTS,
+                    f"Exception when saving crash ({crash_id}); error {errors}/{MAX_ATTEMPTS}"
```

Comment:
The way this loop is implemented, it will fail 20 times in rapid succession and send a bunch of Sentry events. I think it doesn't need to try that hard (20 times is a lot) and I think it should pause between attempts. 

Previously, with the background processing crashmover, it'd get appended to the crashmover queue and there'd be some pause until it was attempted again. Antenna tried 20 times because in that situation, the crash reporter thinks the crash report has been submitted successfully and dropped the crash report data so it was important for Antenna to save it.

There's a `antenna.util.retry` you can use for this loop and the crashpublish loop.

For this loop, I think it should try 5 times with 2 seconds in between iterations. If it's not working, it should give up, return `False`, and then Antenna should return an HTTP 500 error.